### PR TITLE
Add 2deg MOSART grid

### PR DESCRIPTION
**Pull request by rljacob**
Add 2 degree mosart grid, r2_r2 grid alias and allow testing with
--res r2_r2 --compset RMOSGPCC

[BFB]

### CODE REVIEW

**Comment by rljacob** [comment]
I'm also experimenting with the "Draft PR" feature.

**Comment by rljacob** [comment]
@bishtgautam or @hydrotian, can one of you help with this?  I can get a case with "--res r2_r2 --compset RMOSGPCC" to build but it doesn't run.  The error is in the coupler init for datm.
```
(prep_atm_init) : Initializing mapper_Fl2a
(seq_map_init_rcfile)  called for mapper_Fl2a initialization
 (seq_map_gsmapcheck)gsmap global sizes different        55296       18048
 ERROR: (seq_map_gsmapcheck) different gsmap size
```

File: cime_config/config_grids.xml [hunk 1]
```diff
@@ -2784,6 +2794,12 @@
 
     <!-- ROF (river) grids-->
 
+    <domain name="r2">
```
Comment (bishtgautam) [inline cime_config/config_grids.xml:2797]: @rljacob Comparing the `dlnd_in` for ` ERS.r05_r05.RMOSGPCC.cori-knl_intel.mosart-gpcc_1972` test, it appears that there is no entry for land domain file in `dlnd_in` for `--res r2_r2 --compset RMOSGPCC`. I believe you need to specify a grid file here for `atm|lnd`

**Comment by rljacob** [comment]
@jonbob do you know how to get the 2 degree equivalent of SCRIPgrid_0.5x0.5_nomask_c110308.nc ?   Is it derived from the river direction file?

**Comment by jonbob** [comment]
@rljacob - let me look and see

**Comment by jonbob** [comment]
I think I can crank that out -- give me a few minutes

**Comment by jonbob** [comment]
OK, I have something for you to try -- where do you want me to put it?

**Comment by rljacob** [comment]
On blues.  Thanks!

**Comment by rljacob** [comment]
Turns out to get this case to work I'll also need a domain file which needs a mapping file.  I gave you an issue for this  https://acme-climate.atlassian.net/browse/EI-493

**Comment by jonbob** [comment]
OK, I'll test the scrip file then in order to make a mapping file -- and then point you to both. Do we need all the mapping files necessary for ne4pg2-oQU480? So rof2ocn, rof2lnd, atm2rof, lnd2rof (same as atm2rof)?

**Comment by rljacob** [comment]
Yes we'll need all those eventually but not for this particular case.

**Comment by jonbob** [comment]
OK, I have most of the files we'll eventually need and hopefully all the ones you need today on anvil, in the directory /home/ac.jwolfe/r2. I have the following files staged there:
```
domain.lnd.r2_oQU480.210211.nc
domain.ocn.oQU480.210211.nc
domain.ocn.r2_oQU480.210211.nc
map_ne4pg2_to_r2_mono.210211.nc
map_oQU480_to_r2_mono.210211.nc
map_r2_to_ne4pg2_mono.210211.nc
SCRIPgrid_2x2_nomask_c210211.nc
```
Please let me know if they work for you

**Comment by jonbob** [comment]
now also including map_r2_to_oQU480_nn.210211.nc

**Comment by rljacob** [comment]
Thanks that worked!

**Comment by rljacob** [comment]
r05_r05 is already in developer and doesn't take that long so I'll leave out an r2_r2 test.

**Comment by rljacob** [comment]
ERS.r2_r2.RMOSGPCC.chrysalis_intel.mosart-gpcc_1972 passes.

**Review by bishtgautam — APPROVED** [review]

**Pull request merged**